courses/tasks_test.py

```diff
@@ -171,3 +172,81 @@ def test_module_reordering(self):
                 "Example Week 3: Be Social",
             ]
             assert actual == expected
+
+    def test_dont_populate_hidden_modules(self):
+        """
+        If some modules are marked as hidden, don't populate them.
+        """
+        # Find a chapter block
+        first_module_locator_id = None
+        first_module_title = None
+        for block in self.structure_response['blocks'].values():
+            if block['type'] == 'chapter':
+                first_module_locator_id = block['id']
+                first_module_title = block['display_name']
+                break
+
+        response_with_hidden = deepcopy(self.structure_response)
+        response_with_hidden['blocks'][first_module_locator_id][
+            'visible_to_staff_only'] = True
+
+        course = CourseFactory.create()
+        with mock.patch('courses.tasks.requests', autospec=True) as m_req:
+            m_req.get.return_value.status_code = 200
+            m_req.get.return_value.json = lambda: response_with_hidden
+
+            # initial population. Known to work via `test_module_ordering`
+            module_population(course.course_id)
+
+        # Expect all titles except the hidden module's title
+        modules = course.module_set.all().order_by('order')
+        actual = {x.title for x in modules}
+        expected = {
```

Comment:
You could just assert that `first_module_title not in actual`. The others are going to be there, which we assert in the existing tests and it would make it more resilient to changes in underlying test code.